x-pack/plugins/transform/public/app/sections/create_transform/components/aggregation_list/popover_form.tsx

```diff
@@ -266,6 +271,18 @@ export const PopoverForm: React.FC<Props> = ({ defaultData, otherAggNames, onCha
     formValid = validAggName && aggConfigDef.isValid();
   }
 
+  useEffect(() => {
+    if (isPivotAggsConfigWithExtra<unknown>(aggConfigDef)) {
+      try {
+        // @ts-ignore Partial type but we still need to check if parse-able or not
+        JSON.parse(aggConfigDef.aggConfig?.aggTypeConfig?.filterAggConfig);
+        setIsJSONValid(true);
+      } catch (e) {
+        setIsJSONValid(false);
+      }
+    }
+  }, [aggConfigDef]);
+
```

Comment:
There is no need in state here:

```suggestion
 const isJSONValid = useMemo(() => {
  if (isPivotAggsConfigWithExtra<unknown>(aggConfigDef)) {
      try {
        // @ts-ignore Partial type but we still need to check if parse-able or not
        JSON.parse(aggConfigDef.aggConfig?.aggTypeConfig?.filterAggConfig);
        return true;
      } catch (e) {
        return false;
      }
    }
 }, [aggConfigDef])
```

Reply:
Updated here https://github.com/elastic/kibana/pull/137773/commits/4bffcc0de7165fd03a29af129743100f2e5df802